src/uct/sm/mm/base/mm_iface.c

```diff
@@ -173,14 +173,15 @@ static ucs_status_t uct_mm_iface_query(uct_iface_h tl_iface,
 
     status = uct_mm_md_mapper_ops(md)->query(&attach_shm_file);
     ucs_assert_always(status == UCS_OK);
+   
+    /* Only MM transports with attaching to SHM file can support error handling
+     * mechanisms (e.g. EP checking) to check if a peer was down, there is no
+     * safe way to check a process existence (touching a shared memory block of
+     * a peer leads to "bus" error in case of a peer is down) */
     if (attach_shm_file) {
-        /*
-         * Only MM transports with attaching to SHM file can support error
-         * handling mechanisms (e.g. EP checking) to check if a peer was down,
-         * there is no safe way to check a process existence (touching a shared
-         * memory block of a peer leads to "bus" error in case of a peer is
-         * down) */
         iface_attr->cap.flags |= UCT_IFACE_FLAG_EP_CHECK;
+    } else {
+        iface_attr->cap.flags &= ~UCT_IFACE_FLAG_ERRHANDLE_PEER_FAILURE;
```

Comment:
1. maybe set PEER_FAILURE after line 182?
2. why was it needed for keepalive over ud? was shm selected for keepalive?

Reply:
But it's done only when ERROR_HANDLING=y, which is not the default.
ERROR_HANDLING config was intended to override the default decision to disable shared memory transports when error handling is enable. So this config should have priority over the internal logic.

Reply:
discussed offline, xpmem should not be selected when peer failure is tested because of bus error